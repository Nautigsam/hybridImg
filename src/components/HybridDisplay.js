import React from 'react';
import { Link } from 'react-router-dom';
import { Notification, Level, Tag, Image } from 'react-bulma-components';

export const HybridDisplay = ({ hybrid, hideTags }) => {
  return (
    <Notification className="box" style={{ padding: '0.2rem' }}>
      <Link to={`/hybrid/${hybrid.id}`}>
        <Image src={hybrid.url} />
      </Link>
      <Level
        style={{ flexWrap: 'wrap', display: hideTags ? 'none' : 'default' }}
      >
        {hybrid.tags.map(tag => (
          <Tag key={tag.id} color="info">
            {tag.name}
          </Tag>
        ))}
      </Level>
    </Notification>
  );
};

export default HybridDisplay;
