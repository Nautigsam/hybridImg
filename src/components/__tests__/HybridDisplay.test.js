import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { HybridDisplay } from '../HybridDisplay';

import renderer from 'react-test-renderer';

describe('HybridDisplay', () => {
  it('renders correctly', () => {
    const props = {
      hybrid: {
        id: 'H1',
        name: 'Spaghetti/Cataclysme',
        url: './Images/20200409_174234.jpg',
        tags: [
          { id: 'T1', name: 'Sucette' },
          { id: 'T2', name: 'Arc-en-ciel' },
        ],
        grid: 'G1',
        user: 'U1',
      },
    };
    const tree = renderer
      .create(
        <StaticRouter>
          <HybridDisplay {...props} />
          <HybridDisplay {...props} hideLabels />
        </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
