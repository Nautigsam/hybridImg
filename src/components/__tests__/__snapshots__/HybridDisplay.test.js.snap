// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HybridDisplay renders correctly 1`] = `
Array [
  <div
    className="notification box"
    style={
      Object {
        "padding": "0.2rem",
      }
    }
  >
    <a
      href="/hybrid/H1"
      onClick={[Function]}
    >
      <figure
        className="image"
      >
        <img
          alt=""
          className=""
          onError={[Function]}
          src="./Images/20200409_174234.jpg"
        />
      </figure>
    </a>
    <div
      className="level"
      style={
        Object {
          "display": "default",
          "flexWrap": "wrap",
        }
      }
    >
      <span
        className="tag is-info"
      >
        Sucette
      </span>
      <span
        className="tag is-info"
      >
        Arc-en-ciel
      </span>
    </div>
  </div>,
  <div
    className="notification box"
    style={
      Object {
        "padding": "0.2rem",
      }
    }
  >
    <a
      href="/hybrid/H1"
      onClick={[Function]}
    >
      <figure
        className="image"
      >
        <img
          alt=""
          className=""
          onError={[Function]}
          src="./Images/20200409_174234.jpg"
        />
      </figure>
    </a>
    <div
      className="level"
      style={
        Object {
          "display": "default",
          "flexWrap": "wrap",
        }
      }
    >
      <span
        className="tag is-info"
      >
        Sucette
      </span>
      <span
        className="tag is-info"
      >
        Arc-en-ciel
      </span>
    </div>
  </div>,
]
`;
